src/test/java/com/bobocode/hoverla/bring/context/BeanInitializerTest.java

```diff
@@ -0,0 +1,63 @@
+package com.bobocode.hoverla.bring.context;
+
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Stream;
+
+import static java.util.Collections.emptyList;
+import static java.util.Collections.singletonList;
+import static org.mockito.Mockito.*;
+
+class BeanInitializerTest {
+
+    private static final String BD1 = "beanDef1";
+    private static final String BD2 = "beanDef2";
+    private static final String BD3 = "beanDef3";
+    private static final String BD4 = "beanDef4";
+
+    private static Stream<Arguments> testArgs() {
+        BeanDefinition beanDef1 = prepare(BD1, List.of(BD2, BD3));
+        BeanDefinition beanDef2 = prepare(BD2, List.of(BD3, BD4));
+        BeanDefinition beanDef3 = prepare(BD3, singletonList(BD4));
+        BeanDefinition beanDef4 = prepare(BD4, emptyList());
+
+        return Stream.of(
```

Comment:
One suggestion:  don't one-line all this args in one Arguments.of() method. It would be more readable if you assign each of the args to variables and name them accordingly

Reply:
@DanielChanger wrong suggestion :) Each entry of `Arguments.of()` should contain input data and expected results ( bean definition map and expected invocation arguments in our case). Since there are no corner cases, out test has only one entry of arguments.

Reply:
Names are visible in test method parameters :) `Arguments` method name is used in `@MethodSource` annotation. Thus, it is quite obvious, what each parameter is used for. If such approach is not clear for everybody, then I would recommend to avoid the use of `@ParameterizedTest`

Reply:
No....

Reply:
Maybe I wasn't that clear, because we're not on the same page now. Should we have a call?

Reply:
@DanielChanger let’s do it tomorrow after our lesson. I don’t wanna miss it because of lack of sleep 😁

Reply:
It's 5 minutes